cmd/prepare-engine.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"github.com/go-flutter-desktop/hover/internal/build"
+	"github.com/go-flutter-desktop/hover/internal/config"
+	"github.com/go-flutter-desktop/hover/internal/enginecache"
+	"github.com/go-flutter-desktop/hover/internal/log"
+	"github.com/spf13/cobra"
+	"os"
+	"os/exec"
+	"runtime"
+)
+
+var (
+	prepareCachePath     string
+	prepareEngineVersion string
+	prepareReleaseMode   bool
+	prepareDebugMode     bool
+	prepareBuildModes    []build.Mode
+)
+
+func init() {
+	prepareEngineCmd.PersistentFlags().StringVar(&prepareCachePath, "cache-path", enginecache.DefaultCachePath(), "The path that hover uses to cache dependencies such as the Flutter engine .so/.dll")
+	prepareEngineCmd.PersistentFlags().StringVar(&prepareEngineVersion, "engine-version", config.BuildEngineDefault, "The flutter engine version to use.")
+	prepareEngineCmd.PersistentFlags().BoolVar(&prepareDebugMode, "debug-mode", true, "Prepare the flutter engine for debug mode")
+	prepareEngineCmd.PersistentFlags().BoolVar(&prepareReleaseMode, "release-mode", false, "Prepare the flutter engine for release mode")
```

Comment:
For completeness there should also be a profile flag

Reply:
Will do.